### PR TITLE
minor fixes for building on OSX Darwin

### PR DESCRIPTION
**Pull request by khuck**
## Proposed Changes

  - minor change to CMakeLists.txt for link flag
  - std::on_quick_exit() not implemented on OSX, so don't call it.

### CODE REVIEW

**Comment by hkaiser** [comment]
> std::on_quick_exit() not implemented on OSX, so don't call it.

@khuck, this is solved in #4189 in a more portable way.

**Comment by khuck** [comment]
@msimberg  OK, but the APPLE - specific CMake change is still required.  Apple doesn't understand --whole-archive...

**Comment by hkaiser** [comment]
> @msimberg OK, but the APPLE - specific CMake change is still required. Apple doesn't understand --whole-archive...

@khuck: yes, certainly. Could you modify this PR such that the linker flag change remains?

**Comment by khuck** [comment]
@hkaiser @msimberg done

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -328,7 +328,11 @@ foreach(_module ${HPX_LIBS})
   if(UNIX OR APPLE)
     # add the full archive (some are not used directly in libhpx and need to be
     # forced into libhpx with this option)
+  if(APPLE)
+    set(_module_target "-Wl,-all_load" "${_module_target}")
+  else(APPLE)
```
Comment (msimberg) [inline src/CMakeLists.txt]: Sorry for the nitpick. Could we do:
```
if(UNIX)
  set(... whole-archive ...)
elseif(APPLE)
  set(... all_load ...)
endif()
```
instead and skip the nested ifs?

Comment (msimberg) [inline src/CMakeLists.txt]: Sorry for the nitpick. Could we do:
```
if(UNIX)
  set(... whole-archive ...)
elseif(APPLE)
  set(... all_load ...)
endif()
```
instead and skip the nested ifs?

**Comment by msimberg** [comment]
Thanks @khuck, I didn't realize that option would be different on apple.

File: src/CMakeLists.txt (hunk 1 above)
Comment (khuck) [reply src/CMakeLists.txt]: @msimberg done

**Pull request merged**